plugins/rule_engines/irods_rule_language/src/configuration.cpp

```diff
@@ -418,23 +424,54 @@ int hash_rules(const std::vector<std::string> &irbs, const int pid, std::string
     return 0;
 }
 
-class make_copy {
+class in_memory_rulebases {
 public:
-        make_copy(const std::vector<std::string>& _irbs, const int _pid) : irbs_(_irbs), pid_(_pid) {
-            for(auto const &irb : _irbs) {
-                boost::filesystem::copy_file(get_rule_base_path(irb), get_rule_base_path_copy(irb, _pid));
-            }
-        }
-        ~make_copy() {
-            for(auto const &irb : irbs_) {
-                boost::filesystem::remove(get_rule_base_path_copy(irb, pid_));
-            }
-        }
+  in_memory_rulebases(const std::vector<std::string>& _irods_rule_bases)
+      : irods_rule_bases_(_irods_rule_bases)
+  {
+      for (auto const& irb : _irods_rule_bases) {
+          std::ifstream ifs(get_rule_base_path(irb));
+          rule_bases_.insert(
+              {irb, std::string({std::istreambuf_iterator<char>(ifs), std::istreambuf_iterator<char>()})});
+      }
+  }
+
+  const std::string& get_rulebase(const std::string& irb)
+  {
+      return rule_bases_.at(irb);
+  }
+
 private:
-        const std::vector<std::string> irbs_;
-        const int pid_;
+  std::unordered_map<std::string, std::string> rule_bases_;
+  const std::vector<std::string> irods_rule_bases_;
 };
 
+int hash_rules_with_copy(const std::vector<std::string>& irods_rule_bases, std::string& digest, in_memory_rulebases& copy)
+{
+    int status;
+    irods::Hasher hasher;
+    irods::error ret = irods::getHasher("md5", hasher);
+    if (!ret.ok()) {
+        status = ret.code();
+        logger::rule_engine::error("hash_rules_with_copy: cannot get hasher, status = {}", status);
+        return status;
+    }
+
+    for (auto const& irb : irods_rule_bases) {
+        try {
+            hasher.update(copy.get_rulebase(irb));
+        } catch(std::out_of_range& ex) {
+            logger::rule_engine::error("hash_rules_with_copy: attempted to access nonexistent rulebase [{}]", irb);
+        }
+    }
+    
+    hasher.digest(digest);
+
+    logger::rule_engine::debug("hash_rules_with_copy: rulebases = {}, digest = {}", fmt::join(irods_rule_bases, ","), digest);
```

Comment:
Feel free to ignore this since it's just a debug message...

Consider using a semi-colon or something to separate the two items in the log message here since the rulebases are being joined with commas. In other words...
```suggestion
    logger::rule_engine::debug("hash_rules_with_copy: rulebases = {}; digest = {}", fmt::join(irods_rule_bases, ","), digest);
```